docs/units/type_hints.rst

```diff
@@ -0,0 +1,136 @@
+Unit-Aware Type Annotations
+***************************
+
+.. |Quantity| replace:: :class:`~astropy.units.Quantity`
+
+While Python is a dynamically typed language, there is also support for type hints (and even static typing) using the annotations syntax of `PEP 484 <https://www.python.org/dev/peps/pep-0484/>`_.
+For a detailed guide on type hints, function annotations, and other related syntax see the `Real Python Guide <https://realpython.com/python-type-checking/#type-aliases>`_. 
+Here we how Astropy |Quantity| can be used in type hints and annotations and also include metadata about the associated units.
+
+
+For this and all further examples we assume the following imports
+
+::
+
+   >>> import typing as T
+   >>> import astropy.units as u
+   >>> from astropy.units import Quantity
+   >>> from astropy.units._typing import Annotated
+
+
+.. _quantity_type_annotation:
+
+Quantity Type Annotation
+========================
+
+A |Quantity| can easily be used as a type annotation,
+
+::
```

Comment:
Here and below, maybe just append `::` to previous sentence to make more of the file fit on one's screen?

Reply:
Done!. It doesn't look great, but it's not meant to be read in RST form anyway!